Implement/donation section

alirezakariminejad opened the pull request.

alirezakariminejad committed:
refactor(structure): Change archive folder location

alirezakariminejad committed:
chore(deps): Add simple-parallax-js packages

alirezakariminejad committed:
chore(assets): add donation-bg to assets folder

alirezakariminejad committed:
feat(components): add AnimatedCrownSVG.astro component

alirezakariminejad committed:
feat(components): Add ParallaxSection.astro component

alirezakariminejad committed:
refactor(Layout): Add font-pinar class and change text-default to text-regular

alirezakariminejad committed:
refactor(index): Add Donation to index

alirezakariminejad committed:
feat(components): add Donation.astro component

alirezakariminejad committed:
efactor(): Delete the line of code that was not needed

alirezakariminejad committed:
style: add Tailwind classes and improve structure

alirezakariminejad committed:
refactor(cleanup): remove excess code from Donation.astro

alirezakariminejad committed:
chore(types): add ImageMetadata type import to ParallaxSection

alirezakariminejad committed:
refactor(cleanup): remove excess code from LandingLayout.astro

alirezakariminejad committed:
refactor(cleanup): remove excess code from navigation.ts

alirezakariminejad committed:
refactor(cleanup): remove excess code from index.astro

alirezakariminejad committed:
refactor(types): replace any type with proper TypeScript interface in navigation.
BREAKING CHANGE: headerData function now requires translation parameter to match TranslationFunction interface

alirezakariminejad committed:
style: apply prettier formatting across codebase

alirezakariminejad committed:
feat(types): add Donation interface for donation widget

alirezakariminejad committed:
refactor(widget): restructure Donation component to match AstroWind patterns

alirezakariminejad committed:
feat(components): add ParallaxSection component for image backgrounds

alirezakariminejad committed:
feat(pages): add Donation section to landing page

alirezakariminejad committed:
style: run prettier to format code

alirezakariminejad committed:
refactor(cleanup): remove excess code from index.astro

payam-zahedi commented on package.json:
```diff
@@ -33,6 +33,7 @@
     "astro-icon": "^1.1.1",
     "limax": "4.1.0",
     "lodash.merge": "^4.6.2",
+    "simple-parallax-js": "^6.0.1",
```
Nice Package

payam-zahedi commented on src/pages/index.astro:
```diff
@@ -399,4 +400,22 @@ const metadata = {
   </CallToAction>
+
+  <Donation
+    donationText="هر مبلغی، هر چند کوچک، نشانه‌ای از اعتماد شماست و به ما انگیزه می‌دهد تا با قدرت بیشتری به مسیرمان ادامه دهیم. با حمایت شما، می‌توانیم جامعه فلاتر فارسی را به یکی از بهترین منابع آموزشی در دنیای برنامه‌نویسی تبدیل کنیم."
+    donationButtonText="حمایت مالی"
+    donationLink="/"
+    title={{
+      firstPart: 'با حمایت شما، ',
+      secondPart: 'فلاتر فارسی پرواز می‌کند',
+    }}
+    subtitle="جامعه فلاتر فارسی با عشق و تلاش شما رشد کرده و امروز به نقطه‌ای رسیده که می‌تواند منبعی ارزشمند برای هزاران توسعه‌دهنده باشد. حمایت مالی شما به ما کمک می‌کند تا :"
+    features={[
+      'زیرساخت‌های فنی جامعه را بهبود بخشیم',
+      'دوره‌های آموزشی رایگان بیشتری تولید کنیم',
+      'کیفیت محتوای موجود را ارتقا دهیم',
+      'رویدادهای آموزشی بیشتری برگزار کنیم',
+    ]}
+    imageSrc={donationBgImage}
```
I think we don't need to send a parameter to our Donation Component 

Because these parameters are not reusable 

We only use Donation Component in our main page, and it will not change in any other pages or cases 


So let's Remove the parameters and add them to translation files

alirezakariminejad replied:
Fixed in new commit

payam-zahedi commented on src/components/widgets/Donation.astro:
```diff
@@ -0,0 +1,103 @@
+} = Astro.props;
+
+const rotationClasses = [
+  'rotate-0', // برای index 0
+  'rotate-90', // برای index 1
+  '-rotate-90', // برای index 2
+  'rotate-180', // برای index 3
```
Use English comments

alirezakariminejad replied:
Fixed in new commit

payam-zahedi commented on src/components/widgets/AnimatedCrownSVG.astro:
```diff
@@ -0,0 +1,86 @@
+---
+// AnimatedCrownSVG.astro
+---
```
Good job

Thats really nice that you made it separately

payam-zahedi commented on src/components/widgets/Donation.astro:
```diff
@@ -0,0 +1,103 @@
+];
+---
+
+<WidgetWrapper
```
Could you please explain what does this component? WidgetWrapper?

alirezakariminejad replied:
The GitHub link of the tutorial I made with the help of AI 
[https://github.com/alirezakariminejad/astrowind-WidgetWrapper-docs](url)

payam-zahedi commented on src/components/widgets/ParallaxSection.astro:
```diff
@@ -0,0 +1,34 @@
+  import simpleParallax from 'simple-parallax-js/vanilla';
+
+  function setupParallax() {
+    const images = document.querySelectorAll('.parallax-image');
```
Nice Job 

Its really Dope

payam-zahedi commented on src/components/widgets/Donation.astro:
```diff
@@ -0,0 +1,103 @@
+>
+  <div class="absolute w-full h-full">
+    <span id="color-layer" class="absolute left-0 top-0 w-full h-full bg-[#ebebeb] -z-10"></span>
+    {imageSrc && <ParallaxSection imageSrc={imageSrc} altText="تصویر پارالاکس" />}
```
Use translation for altText

alirezakariminejad replied:
Fixed in new commit

payam-zahedi reviewed (approved):
Well Done @alirezakariminejad

alirezakariminejad committed:
refactor(components): move translations inside Donation component

alirezakariminejad committed:
refactor(types): simplify Donation interface with minimal props
- Remove translation-related props from interface
- Keep only essential configuration props

alirezakariminejad committed:
feat(i18n): add Donation component translations

alirezakariminejad committed:
refactor(pages): simplify Donation component usage in index page
- Remove hardcoded translation strings
- Simplify component props to minimal configuration
- Keep only donationLink as direct prop

alirezakariminejad committed:
feat(pages): add Donation section to English landing page

alirezakariminejad committed:
refactor(components): Remove type TFunction

alirezakariminejad committed:
style: run prettier to format code

alirezakariminejad committed:
feat(i18n): add altText to Donation component translations

alirezakariminejad committed:
feat(Donation): add altText to Translations

alirezakariminejad committed:
refactor(ParrallaxSection): remove default altText value

alirezakariminejad committed:
refactor(cleanup): remove excess code from index.astro

alirezakariminejad committed:
feat(Donation): add dark mode support to Donation component

alirezakariminejad committed:
refactor(types): extend Donation interface from Widget base type

alirezakariminejad committed:
feat(AnimatedCrownSVG): change hover animation to crown icon

alirezakariminejad committed:
feat(types): move Donation interface to component file

alirezakariminejad committed:
refactor(donation): optimize type imports

alirezakariminejad committed:
fix(donation): adjust box order in mobile and desktop views

alirezakariminejad committed:
refactor(donation): update ImageMetadata import path

alirezakariminejad committed:
style: run prettier to format code

payam-zahedi committed:
fix: fix text alignment for donation component in english version
